napari/utils/events/containers/_nested_list.py

```diff
@@ -0,0 +1,398 @@
+    reordered (value: self)
+        emitted when the list is reordered (eg. moved/reversed).
+    """
+
+    # WAIT!! ... Read the ._list module docs before reimplement these classes
+    # def append(self, item): ...
+    # def clear(self): ...
+    # def pop(self, index=-1): ...
+    # def extend(self, value: Iterable[_T]): ...
+    # def remove(self, value: T): ...
+
+    @overload
+    def __getitem__(self, key: int) -> _T:
+        ...  # pragma: no cover
+
+    @overload
+    def __getitem__(  # noqa: F811
+        self, key: ParentIndex
+    ) -> 'NestableEventedList[_T]':
+        ...  # pragma: no cover
+
+    @overload
+    def __getitem__(self, key: slice) -> 'NestableEventedList[_T]':  # noqa
+        ...  # pragma: no cover
+
+    @overload
+    def __getitem__(  # noqa: F811
+        self, key: NestedIndex
+    ) -> Union[_T, 'NestableEventedList[_T]']:
+        ...  # pragma: no cover
+
+    def __getitem__(self, key):  # noqa: F811
+        if isinstance(key, tuple):
+            item: NestableEventedList[_T] = self
+            for idx in key:
+                item = item[idx]
+            return item
+        return super().__getitem__(key)
+
+    @overload
+    def __setitem__(self, key: Union[int, NestedIndex], value: _T):
+        ...  # pragma: no cover
+
+    @overload
+    def __setitem__(self, key: slice, value: Iterable[_T]):  # noqa: F811
+        ...  # pragma: no cover
+
+    def __setitem__(self, key, value):  # noqa: F811
+        # NOTE: if we check isinstance(..., MutableList), then we'll actually
+        # clobber object of specialized classes being inserted into the list
+        # (for instance, subclasses of NestableEventedList)
+        # this check is more conservative, but will miss some "nestable" things
+        if isinstance(value, list):
+            value = self.__class__(value)
+        if isinstance(key, tuple):
+            parent_i, index = split_nested_index(key)
+            self[parent_i].__setitem__(index, value)
+            return
+        self._connect_child_emitters(value)
+        super().__setitem__(key, value)
+
+    @overload
+    def _delitem_indices(
+        self, key: Index
+    ) -> Iterable[Tuple[EventedList[_T], int]]:
+        ...  # pragma: no cover
+
+    @overload
+    def _delitem_indices(  # noqa: F811
+        self, key: NestedIndex
+    ) -> Iterable[Tuple[EventedList[_T], Index]]:
+        ...  # pragma: no cover
+
+    def _delitem_indices(self, key):  # noqa: F811
+        if isinstance(key, tuple):
+            parent_i, index = split_nested_index(key)
+            return [(cast(NestableEventedList[_T], self[parent_i]), index)]
+        return super()._delitem_indices(key)
+
+    def __delitem__(self, key):
+        # delete from the end
+        for parent, index in self._delitem_indices(key):
+            self._disconnect_child_emitters(parent[index])
+        super().__delitem__(key)
+
+    # TODO: implement __eq__
```

Comment:
```suggestion
```

Can we replace this comment by an issue? =)

Reply:
will delete for now.  I'm actually not sure we _need_ to implement that.  It will become obvious if we do :)